src/test/java/de/tum/in/www1/artemis/service/connectors/LtiServiceTest.java

```diff
@@ -65,11 +73,22 @@
 
     private LtiOutcomeUrl ltiOutcomeUrl;
 
+    private HttpClient httpClient;
+
+    private HttpResponse httpResponse;
+
+    private StatusLine statusLine;
+
     @BeforeEach
     public void init() {
         MockitoAnnotations.openMocks(this);
         SecurityContextHolder.clearContext();
+        httpClient = mock(HttpClient.class);
+        httpResponse = mock(HttpResponse.class);
+        statusLine = mock(StatusLine.class);
         ltiService = new LtiService(userCreationService, userRepository, ltiOutcomeUrlRepository, resultRepository, artemisAuthenticationProvider, ltiUserIdRepository, response);
+        ltiServiceWithClient = new LtiService(userCreationService, userRepository, ltiOutcomeUrlRepository, resultRepository, artemisAuthenticationProvider, ltiUserIdRepository,
```

Comment:
Ideally LtiService doesn't need to created, but can also be injected using @Autowired above (which is the default solution for injecting services into tests)

Reply:
removed